### PR TITLE
build: update version-bumper to support alpha

### PR DESCRIPTION
**Pull request by VerteDinde**
#### Description of Change

Updates the version-bumper script and supporting utils to support an alpha build for Electron 15.

This PR also adds a mock to support new version-bumper tests for alpha tags. Because we don't currently have alpha tags and were previously using Electron's actual git tags from 2.x.x, a mock was required to test for alphas. These tests are currently separated from the rest to make them easier to delete, but they can easily be integrated with the main tests if we decide we want to keep the alpha channel.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

**Comment by VerteDinde** [comment]
Max & Linux failures are unrelated, but it looks like my alpha tests are causing memory to run out on Windows 🤔 Looking into a fix now

File: script/release/prepare-release.js [hunk 1]
```diff
@@ -182,7 +187,7 @@ async function promptForVersion (version) {
 
 // function to determine if there have been commits to main since the last release
 async function changesToRelease () {
-  const lastCommitWasRelease = new RegExp('^Bump v[0-9.]*(-beta[0-9.]*)?(-nightly[0-9.]*)?$', 'g');
+  const lastCommitWasRelease = new RegExp('^Bump v[0-9.]*(-beta[0-9.]*)?(-alpha[0-9.]*)?(-nightly[0-9.]*)?$', 'g');
```
Comment (MarshallOfSound) [inline script/release/prepare-release.js:190]:
```suggestion
  const lastCommitWasRelease = new RegExp('^Bump v[0-9]+\.[0-9]+\.[0-9]+(-beta\.[0-9]+)?(-alpha\.[0-9]+)?(-nightly\.[0-9]+)?$', 'g');
```

File: script/release/publish-to-npm.js [hunk 2]
```diff
@@ -175,6 +175,10 @@ new Promise((resolve, reject) => {
             semver.gt(localVersion, currentTags.beta)) {
         childProcess.execSync(`npm dist-tag add electron@${localVersion} beta --otp=${process.env.ELECTRON_NPM_OTP}`);
       }
+      if (parsedLocalVersion.prerelease[0] === 'alpha' &&
+            semver.gt(localVersion, currentTags.alpha)) {
+        childProcess.execSync(`npm dist-tag add electron@${localVersion} alpha --otp=${process.env.ELECTRON_NPM_OTP}`);
```
Comment (MarshallOfSound) [inline script/release/publish-to-npm.js:180]: We also have to make changes on line 140~ish to make the tag `alpha-${currentBranch}`

Comment (VerteDinde) [reply script/release/publish-to-npm.js:183]: I added a check for the alpha NPM tag on 138:
```
} else if (release.tag_name.indexOf('alpha') > 0) {
```

I double-checked the GitHub tag system, but I'm a little wary of whether or not this check will work as expected.

File: script/release/prepare-release.js (hunk 1 above)
Comment (VerteDinde) [reply script/release/prepare-release.js]: Looks like es-lint thinks these aren't needed! Added an eslint-ignore comment to this, but can also revert it

**Review by MarshallOfSound — APPROVED** [review]

**Pull request merged**